### PR TITLE
Multiplatform docker

### PR DESCRIPTION
**Pull request by hilli**
# Description

I am doing development on an Apple Silicon Mac and doing deploys to other ARM based hosts. I have previously build the images myself, but figured that I can't be the only one who would like hanko on more than one architecture.

# Implementation

- Added parameter in the Dockerfile's to enable building for multiple architectures
  - Fixing the builder to the BUILDPLATFORM
  - Utilizing GOs crossplatform builder to target TARGETARCH
  - Let `buildx` handle the matrix (1x2, but not the less :)) from there
- Bumped Docker's Github Actions versions to get rid of warnings, hopefully
- Bumped the go version to 1.19 in the Dockerfile

# Tests

Try running the action related to the workflow on this branch. There should be bulds for both amd64 and arm64 on ghcr.io afterwards.

# Todos

There are other architectures available (linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6), which could be added to the matrix, but I would wait with that until there is a need for it.

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -48,15 +48,17 @@ jobs:
...
+
       - name: Setup Docker buildx
-        uses: docker/setup-buildx-action@v1.6.0
+        uses: docker/setup-buildx-action@v2
 
       # Login against a Docker registry except on PR
       # https://github.com/docker/login-action
       - name: Log into registry ${{ env.REGISTRY }}
         if: github.event_name != 'pull_request'
-        uses: docker/login-action@v1.14.1
```
Comment (like-a-bause) [inline .github/workflows/docker-publish.yml:59]: Did you want to update to v2 ?

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -48,15 +48,17 @@ jobs:
...
+        uses: docker/login-action@v1.2
```
Comment (like-a-bause) [inline .github/workflows/docker-publish.yml:61]: I wanted to mark this line. :)

Did you want to update to v2 ?

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (hilli) [reply .github/workflows/docker-publish.yml:59]: Same here.

Comment (hilli) [reply .github/workflows/docker-publish.yml:59]: Cause it's the same place ;)

**Review by like-a-bause — APPROVED** [review]
Awesome! Thanks a lot!

**Pull request merged**